pulp_container/app/token_verification.py

```diff
@@ -127,10 +128,14 @@ def has_permission(self, request, view):
         """
         Decide upon permission based on token
         """
+        decoded_token = request.auth
+        if decoded_token is True:
+            return True
+
         try:
-            decoded_token = request.auth
-            return decoded_token is True or _contains_accessible_actions(
-                decoded_token, *_access_scope(request)
-            )
+            if _contains_accessible_actions(decoded_token, *_access_scope(request)):
+                return True
         except Exception:
-            return False
+            raise NotAuthenticated()
+
+        raise AuthenticationFailed(code="insufficient_scope")
```

Comment:
if you don't add a `detail` here the user will get ` incorrect username/password` which is misleading
can you add `detail=insufficient permissions` or something along the lines